vpdq/benchmark/benchmark.py

```diff
@@ -0,0 +1,151 @@
+from numpy import hamming
+import vpdq
+import subprocess
+import os
+import sys
+import argparse
+from pathlib import Path
+from contextlib import contextmanager, nullcontext
+from enum import Enum
+import time
+
+VPDQ_MATCH_DIST = 31
+DIR = Path(__file__).parents[2]
+VIDEOS = DIR / "tmk/sample-videos"
+OUTPUT = "hashes"
+CPP_EXEC = Path(__file__).parents[1] / "cpp/build/vpdq-hash-video"
+
+
+@contextmanager
+def timer(context: str, print_on_enter: bool = False):
+    if print_on_enter:
+        print(f"{context}...")
+    start = time.perf_counter()
+    end = start
+    yield lambda: end - start
+    end = time.perf_counter()
+    print(f"{context}: {end - start:.4f}s")
+
+
+def get_argparse() -> argparse.ArgumentParser:
+    ap = argparse.ArgumentParser(description=__doc__)
+    ap.add_argument(
+        "-f",
+        "--ffmpegPath",
+        help="Specific path to ffmpeg you want to use",
+        default="ffmpeg",
+    )
+    ap.add_argument(
+        "-r",
+        "--secondsPerHash",
+        help="The frequence(per second) a hash is generated from the video. If it is 0, will generate every frame's hash",
+        default=0,
+        type=float,
+    )
+    ap.add_argument(
+        "-s",
+        "--downsampleFrameDimension",
+        metavar="Downsample_Frame_Dimension",
+        help="Resolution to downsample the video to before hashing frames.. If it is 0, will use the original dimension of the video to hash",
+        default=0,
+        type=int,
+    )
+    return ap
+
+
+def run_benchmark(
+    ffmpegPath: str,
+    secondsPerHash: int,
+    downsampleFrameDimension: int,
+):
+    with timer("Python hashing time") as pt:
+        python_features = run_python(
+            ffmpegPath, secondsPerHash, downsampleFrameDimension
+        )
+    feature_size = 0
+    for feature in python_features:
+        feature_size += len(feature)
```

Comment:
ignorable/code golf: 
```suggestion
    feature_size = sum(len(feature) for feature in python_features)```